test/jdk/javax/net/ssl/SSLSession/TestEnabledProtocols.java

```diff
@@ -141,16 +137,25 @@ protected void runClientApplication(SSLSocket sslSocket) throws Exception {
             }
         } catch (SSLException ssle) {
             // The server side may have closed the socket.
-            System.out.println("Client SSLException:");
-            ssle.printStackTrace(System.out);
+            if ("Connection reset".equals(ssle.getCause().getMessage())) {
+                System.out.println("Client SSLException:");
+                ssle.printStackTrace(System.out);
```

Comment:
An SSLException doesn't necessarily have a nested `cause` - so you could get a NPE here.
I would suggest moving that code to a `private boolean isConnectionReset(SSLException ssle)` method, and possibly checking the type of the `cause` exception as well. I'm guessing it should be a `SocketException`?
If you check the type then it will also exclude the case were the cause is null and avoid the NPE.
Additional note: as a general rule, relying on exception message is fragile. But since we don't have a specific subtype for "Connection reset" it's probably the best we can do.

Reply:
Yes, added the verification to check if it is instanceof SocketException